pkg/mmr/memstorage.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package mmr
+
+import (
+	"github.com/tidwall/btree"
+)
+
+type MemStorage struct {
```

Comment:
I think a little description here would be useful. At least describing the contents of the map (I had to double-check what the key was)